test/Faker/Provider/de_AT/PersonTest.php

```diff
@@ -36,4 +36,35 @@ public function testSsn()
             $this->assertLessThanOrEqual(999, $consecutiveNumber);
         }
     }
+
+    public function testSsnWithGivenBirthDateCorrect()
+    {
+        $number = $this->faker->ssn(new \DateTime('2010-09-08 06:05:04'));
+
+        $correctBirthDateString = '080910';
+        $this->assertEquals($correctBirthDateString, substr($number, 4, 6));
+
+        // Should have 10 characters: 3 consecutive number, 1 verification number, 6 birth date
+        $this->assertMatchesRegularExpression('/^[1-9]{1}[0-9]{9}$/', $number, '');
+
+        // consecutive number must be between 100 and 999
+        $consecutiveNumber = substr($number, 0, 3);
+        $this->assertGreaterThanOrEqual(100, $consecutiveNumber);
+        $this->assertLessThanOrEqual(999, $consecutiveNumber);
+
+        // Verification number is correct
+        $verificationNumber = substr($number, 3, 1);
+        $correctVerificationNumber = (
```

Comment:
You can use an array to simplify the calculation:
```php
$weights = [3, 7, 9, 0, 5, 8, 4, 2, 1, 6];
$checkSum = 0;
foreach ($weights as $i => $weight) {
$checkSum += (int) $number[$i] * $weight;
}

self::assertEquals($number[3], $checkSum % 11);
```

You may also want to extract this logic to a separate method like `assertValidSsn` and reuse in the other test method.

Reply:
Alright, added the verification number check the other test method too.